files/collect_api_events.rb

```diff
@@ -6,29 +6,44 @@ def require_classes(modulepaths)
   catch :done do
     modulepaths.split(':').each do |modulepath|
       Find.find(modulepath) do |path|
-        if path.match?(%r{common_events_library.gemspec})
+        if %r{common_events_library.gemspec}.match?(path)
           $LOAD_PATH.unshift("#{File.dirname(path)}/lib")
           throw :done
         end
       end
     end
   end
 
+  require 'events_collection/lockfile'
   require 'events_collection/orchestrator_event'
   require 'common_events_library'
 end
 
-def main(confdir, modulepaths)
+def main(confdir, modulepaths, statedir)
   require_classes(modulepaths)
 
-  settings = YAML.safe_load(File.read("#{confdir}/events_collection.yaml"))
-
-  orchestrator_client = Orchestrator.new('localhost', username: settings['pe_username'], password: settings['pe_password'], token: settings['pe_token'], ssl_verify: false)
-  puts orchestrator_client.get_jobs(limit: 1)
+  begin
+    lockfile = CommonEvents::Lockfile.new(statedir)
+    settings = YAML.safe_load(File.read("#{confdir}/events_collection.yaml"))
+    if lockfile.already_running?
+      puts 'already running'
+    else
+      lockfile.write_lockfile
+      orchestrator_client = Orchestrator.new('localhost', username: settings['pe_username'], password: settings['pe_password'], token: settings['pe_token'], ssl_verify: false)
+      puts orchestrator_client.get_jobs(limit: 1)
+      # Find any compatible reports
+      # Reports should be in /lib/reports/common_events
+    end
+  rescue => exception
+    puts exception
+  ensure
+    lockfile.remove_lockfile
+  end
 end
 
 if $PROGRAM_NAME == __FILE__
   confdir     = ARGV[0]
   modulepaths = ARGV[1]
-  main(confdir, modulepaths)
+  statedir    = ARGV[2]
```

Comment:
Are there reasonable defaults that we can `||` into these input params?

Reply:
I added a few defaults, lmk what you think.

Reply:
Is it possible to get the confdir and modulepath from puppet itself? Not a fan of hardcoded paths.